File: training/algorithm/1_1_1.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
##################################
# University of Wisconsin-Madison
# Author: Yaqi Zhang
##################################
# This module contains
##################################

import itertools

if __name__ == "__main__":
    def to_num(a, b, c):
        return 100 * a + 10 * b + c
    lst = range(1, 10)
    for nums in itertools.permutations(lst):
        x = to_num(*nums[:3])
        y = to_num(*nums[3:6])
        z = to_num(*nums[6:])
        if y == 2 * x and z == 3 * x:
            print("{}{}{}".format(x,y,z))

File: training/algorithm/1_1_1.cpp
/*
 * written by Yaqi Zhang (zhang623@wisc.edu)
 * University of Wisconsin-Madison
 * May 2018
 */

#include <iostream>
#include <algorithm>
using namespace std;

int main()
{
    int x, y, z;
    int d[9] = {1, 2, 3, 4, 5, 6, 7, 8, 9};
    do {
        x = d[0] * 100 + d[1] * 10 + d[2];
        y = d[3] * 100 + d[4] * 10 + d[5];
        z = d[6] * 100 + d[7] * 10 + d[8];
        if (y == x * 2 && z == x * 3) {
            cout << x << y << z << endl;
        }
    } while (next_permutation(d, d + 9));
    return 0;
}

